### PR TITLE
read env file

### PR DESCRIPTION
**Pull request by ericsciple**

### CODE REVIEW

File: src/Runner.Worker/GitHubContext.cs [hunk 1]
```diff
@@ -39,5 +39,17 @@ public sealed class GitHubContext : DictionaryContextData, IEnvironmentContextDa
                 }
             }
         }
+
```
Comment (ericsciple) [inline src/Runner.Worker/GitHubContext.cs:42]: @thboop i'll rebase after your PR is in

File: src/Test/L0/Worker/Handlers/OutputManagerL0.cs [hunk 2]
```diff
@@ -14,7 +14,7 @@
 using Xunit;
 using DTWebApi = GitHub.DistributedTask.WebApi;
 
-namespace GitHub.Runner.Common.Tests.Worker
+namespace GitHub.Runner.Common.Tests.Worker.Handlers
```
Comment (ericsciple) [inline src/Test/L0/Worker/Handlers/OutputManagerL0.cs:17]: revert

File: src/Runner.Worker/Handlers/CompositeActionHandler.cs [hunk 3]
```diff
@@ -73,8 +62,13 @@ public async Task RunAsync(ActionRunStage stage)
 
                 var step = ExecutionContext.CreateCompositeStep(childScopeName, actionRunner, inputsData, Environment);
 
+                // Shallow copy github context
+                var gitHubContext = step.ExecutionContext.ExpressionValues["github"] as GitHubContext;
```
Comment (thboop) [inline src/Runner.Worker/Handlers/CompositeActionHandler.cs:66]: looks like this change may have broken an l0 test, we should track that down

File: src/Runner.Worker/Handlers/OutputFileUtil.cs [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+                var index = 0;
+                var line = ReadLine(text, ref index);
+                while (line != null)
+                {
+                    if (!string.IsNullOrEmpty(line))
+                    {
+                        var equalsIndex = line.IndexOf("=", StringComparison.Ordinal);
+                        var heredocIndex = line.IndexOf("<<", StringComparison.Ordinal);
+
+                        // Normal style NAME=VALUE
```
Comment (thboop) [inline src/Runner.Worker/Handlers/OutputFileUtil.cs:24]: Great comments !

**Review by tomasreyes — COMMENTED** [review]
H

**Review by tomasreyes — COMMENTED** [review]
H

**Pull request merged**